pkg/dispatcher/config.go

```diff
@@ -90,6 +123,23 @@ func (config *Config) DetermineClusterForJob(jobBase prowconfig.JobBase, path st
 		if cluster, ok := jobBase.Labels[api.ClusterLabel]; ok {
 			return api.Cluster(cluster), false, nil
 		}
+	}
+
+	if config.DetermineE2EByJob {
+		if cloud := DetermineCloud(jobBase); cloud != "" {
+			if clusters, ok := config.BuildFarm[api.Cloud(cloud)]; ok {
+				l := len(clusters)
+				keys := make([]string, 0, l)
+				for k := range clusters {
+					keys = append(keys, string(k))
+				}
+				sort.Strings(keys)
+				return api.Cluster(keys[len(jobBase.Name)%l]), false, nil
```

Comment:
The list of clusters (keys) for the given cloud looks like something that would be great to precompute in the config object itself rather than recomputing it for each job.

Reply:
I will give it a try later today.

Reply:
Done.